test/Atmos/EDMF/edmf_kernels.jl

```diff
@@ -733,10 +643,9 @@ function source(::GradProdSource{en_ρaθ_liq_q_tot_cv}, atmos, args)
 end
 
 function source(::BuoySource{up_ρaw{i}}, atmos, args) where {i}
-    # TODO: Cache buoyancy
+    @unpack buoy = args.precomputed.turbconv
     up = args.state.turbconv.updraft
-    up_aux = args.aux.turbconv.updraft
-    return up[i].ρa * up_aux[i].buoyancy
+    return up[i].ρa * buoy.up[i]
```

Comment:
Should this tendency also be wrapped in `fix_void_up`>

Reply:
I’ll let @yairchn address this in #1927

Reply:
#1927 is still a draft PR - should I make it a PR ?

Reply:
also buoy.up[I] is computed from ts.up which is protected with up[i].ρa

Reply:
We’ll probably need to rebase #1927 once this merges